test/spec/tokenManager.js

```diff
@@ -107,6 +107,43 @@ describe('TokenManager', function() {
       client.emitter.emit('fake', payload);
       expect(handler).not.toHaveBeenCalled();
     });
+
+    it('should emit too many renew error when too many renews happen in a short peroid of time', () => {
+      expect.assertions(9);
+      setupSync({
+        tokenManager: { autoRenew: true }
+      });
+      const handler = jest.fn().mockImplementation(err => {
+        util.expectErrorToEqual(err, {
+          name: 'AuthSdkError',
+          message: 'Too many token renew requests',
+          errorCode: 'INTERNAL',
+          errorSummary: 'Too many token renew requests',
+          errorLink: 'INTERNAL',
+          errorId: 'INTERNAL',
+          errorCauses: []
+        });
+      });
+      client.tokenManager.on('error', handler);
+      client.tokenManager.renew = jest.fn().mockImplementation(() => Promise.resolve());
+      for (let i = 0; i <= 10; i++) {
+        client.emitter.emit('expired');
```

Comment:
I think it will emit an error on the 10th request. But what happens on the 11th?

Reply:
it will keep emitting the error. Maybe should add logic to re-evaluate the flag and also make sure only one error is sent out if the evaluation is the same as the previous one.